Remove backward ops for cuDNN transposed convolution

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69902
* #69901

Differential Revision: [D33093795](https://our.internmc.facebook.com/intern/diff/D33093795)

jbschlosser committed:
Remove backward ops for cuDNN transposed convolution

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33093795).

albanD commented on torch/csrc/autograd/FunctionsManual.cpp:
```diff
@@ -4763,6 +4763,22 @@ std::tuple<Tensor, Tensor> _cudnn_convolution_backward(
   return result;
 }
 
+std::tuple<Tensor, Tensor> _cudnn_convolution_transpose_backward(
```
Can we re-use the op from above with an extra boolean argument for transpose?

albanD reviewed (commented):
I think we should re-use the other cudnn backward op. Otherwise looks good.

jbschlosser committed:
Update on "Remove backward ops for cuDNN transposed convolution"


Differential Revision: [D33093795](https://our.internmc.facebook.com/intern/diff/D33093795)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33093795).

albanD reviewed (approved):
SGTM

jbschlosser committed:
Update on "Remove backward ops for cuDNN transposed convolution"


Differential Revision: [D33093795](https://our.internmc.facebook.com/intern/diff/D33093795)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33093795).

albanD reviewed (approved):
Good catch

jbschlosser committed:
Update on "Remove backward ops for cuDNN transposed convolution"


Differential Revision: [D33093795](https://our.internmc.facebook.com/intern/diff/D33093795)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33093795).

jbschlosser committed:
Update on "Remove backward ops for cuDNN transposed convolution"


Differential Revision: [D33093795](https://our.internmc.facebook.com/intern/diff/D33093795)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33093795).

jbschlosser committed:
Update on "Remove backward ops for cuDNN transposed convolution"


Differential Revision: [D33093795](https://our.internmc.facebook.com/intern/diff/D33093795)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33093795).
